src/bin/pgcopydb/lock_utils.c

```diff
@@ -208,7 +208,14 @@ semaphore_unlink(Semaphore *semaphore)
 
 	semun.val = 0;              /* unused, but keep compiler quiet */
 
-	log_debug("ipcrm -s %d", semaphore->semId);
+	/*
+	 * log has dependency on semaphore, so we can't use log_debug() here.
+	 * We use fformat() instead to avoid crash while exiting.
+	 */
```

Comment:
Well the semaphore has not been deleted yet, and the `log_debug` function is sync (returns when the logging has been done already), so I'm not sure I follow?

Reply:
Wrong call!. I just fixed the actual issue in the new commit. PTAL.